src/v/cloud_storage/tests/util.cc

```diff
@@ -564,7 +564,6 @@ std::vector<in_memory_segment> replace_segments(
         auto bo = s.base_offset;
         auto it = manifest.find(bo);
         BOOST_REQUIRE(it != manifest.end());
-        BOOST_REQUIRE(it->size_bytes != s.bytes.size());
```

Comment:
This change makes sense to me, since the batches generated are random. Out of curiosity, was the switch to replacing with `merged_batch_types` required also? Was there another failure mode in this test aside from this size check here?